lib/src/fnc/array.rs

```diff
@@ -105,6 +105,29 @@ pub fn union(arrays: (Value, Value)) -> Result<Value, Error> {
 	})
 }
 
+pub fn insert((array, data, index): (Value, Value, Value)) -> Result<Value, Error> {
+	match (array, index) {
+		(Value::Array(mut v), Value::Number(index)) => {
+			let mut index = index.as_int();
+			if index < 0 {
+				// negative index means start from the back
+				index += v.len() as i64;
+			}
+			if index > v.len() as i64 || index < 0 {
+				// insert will panic
```

Comment:
I don't think the entire array should be emptied if the index is too large. I personally think that if index exceeds the length, it should either:

1. return the original array unaltered
2. add the item to the end of the array

Thoughts on this @finnbear?

Reply:
i think for any index apart from -length to + length . We should return the array unaltered.